### DISCUSSION TITLE
I want to use Component has unique atom value.

### DISCUSSION THREAD

**Discussion by wotjd1799** [pmndrs/jotai#1364] [Q&A] [answered]
We want to manage `atom` values by component.
For example, If you want to create a `TexBox` component, and let each component have the unique `atom` value.
However, the current code shares the `atom` value each other.
Is this possible without using a `<Provider>`?
If I have to use a `<Provider>`, I want to use a provider inside each component.

**example**
```
import { useAtom, atom } from "jotai";

const baseAtom = atom("");
const textAtom = atom(
  (get) => get(baseAtom),
  (get, set, update) => {
    set(baseAtom, update);
  }
);

const TextBox = () => {
  const [text, setText] = useAtom(textAtom);

  const setData = (e) => {
    if (e) {
      setText(e.target.value);
    }
  };

  return (
    <>
      <span className="input-text" style={{ width: "100%" }}>
        <div>text: {text}</div>
        <input
          type="text"
          title="title"
          name="title"
          value={text}
          onChange={setData}
        />
      </span>
    </>
  );
};

export default TextBox;
```

```
...
class SomePage {
  render() {
    <TextBox />
    <TextBox />
  }
}
...
```

**Comment by dai-shi** [comment, marked as answer]
If `Provider` works, it's probably the best usage. (If not, `scope` might help. If not, atoms-in-atom and/or 3rd-party lib would be required.)

Check this: https://twitter.com/dai_shi/status/1447892237753278466

**Reply by dongwooklee96** [reply]
@dai-shi 
In these cases, how can I get each different status value?

**Reply by dai-shi** [reply]
You get different values in different providers. I'm not sure if I get the question correctly, but let me show useState+useContext solution, because essentially, it's the same.

https://codesandbox.io/s/cranky-microservice-hmi2l3?file=/src/App.js

```jsx
import { createContext, useContext, useState } from "react";

const CounterContext = createContext();

const Provider = ({ children }) => {
  const [count, setCount] = useState(0);
  return (
    <CounterContext.Provider value={[count, setCount]}>
      {children}
    </CounterContext.Provider>
  );
};

const Counter = () => {
  const [count, setCount] = useContext(CounterContext);
  const inc = () => setCount((c) => c + 1);
  return (
    <div>
      {count} <button onClick={inc}>+1</button>
    </div>
  );
};

const App = () => (
  <>
    <Provider>
      <h1>Provider 1</h1>
      <Counter />
      <Counter />
    </Provider>
    <Provider>
      <h1>Provider 2</h1>
      <Counter />
      <Counter />
    </Provider>
  </>
);

export default App;
```

**Reply by wotjd1799** [reply]
- in this situation, how can i get these values in SomePage Component?
![스크린샷 2022-08-24 오전 10 39 55](https://user-images.githubusercontent.com/65709575/186299328-21c644af-46d8-4aa8-baf3-92229e4494f5.png)

- And, if i want to get the value without setting the form, such as storing the value in atom. 
Because I want to manage data freely.
if I have to call the api from the 'Somepage' component using all the input values, how can I use it?

```Javascript
const SomePage = () => {
  // I want called api using Input Values.

  return (
    <>
      <Provider>
        <TextBox />
      </Provider>
      <Provider>
        <TextBox />
      </Provider>
      <Provider>
        <BooleanBox />
      </Provider>
      <Provider>
        <BooleanBox />
      </Provider>
    </>
  );
};

export default SomePage;
```

**Reply by dai-shi** [reply]
Sorry, I'm not sure if I get the question. Can you create a codesandbox for one question?

In general, if you can do with useState+useContext, it can be done with Jotai just like that. (1st option)
Does you use case work with useState+useContext without Jotai?

**Reply by wotjd1799** [reply]
https://codesandbox.io/s/vigorous-cloud-lsoycv?file=/src/App.js
If I'm trying to use jotai in the wrong way, please tell me!

**Reply by dai-shi** [reply]
So you want have a common `TextBox` component, and you want all values in `App`.
For this, you would need to pass something to TextBox from App, because it's contextual.
Provider doesn't work well in this use case, because atom values only exist inside Provider.

I would pass an atom to TextBox. Simple version is here:
https://codesandbox.io/s/suspicious-wright-6oqrrw?file=/src/App.js

See atoms-in-atom for more patterns: https://jotai.org/docs/guides/atoms-in-atom

**Reply by wotjd1799** [reply]
Oh, I didn't think of the basics. Thank you very much.
